values.yaml

```
@@ -328,10 +328,24 @@ syncCatalog:
   # to all namespaces.
   k8sSourceNamespace: null
 
-  # consulPrefix is the service prefix which preprends itself
+  # addK8SNamespaceSuffix appends Kubernetes namespace suffix to
+  # each service name synced to Consul, separated by a dash.
+  # For example, for a service 'foo' in the default namespace,
+  # the sync process will create a Consul service named 'foo-default'.
+  # Set this flag to true to avoid registering services with the same name
+  # but in different namespaces as instances for the same Consul service.
+  # Namespace suffix is not added if 'annotationServiceName' is provided.
+  # NOTE: Setting this property to 'true' will result in deregistering of existing
+  # synced services in Consul and registering them with a new name.
+  addK8SNamespaceSuffix: true
+
+  # consulPrefix is the service prefix which prepends itself
   # to Kubernetes services registered within Consul
-  # For example, "k8s-" will register all services peprended with "k8s-".
+  # For example, "k8s-" will register all services prepended with "k8s-".
   # (Kubernetes -> Consul sync)
+  # consulPrefix is ignored when 'annotationServiceName' is provided.
+  # NOTE: Setting this property to a non-null value will result in deregistering
```

```suggestion
  # NOTE: Updating this property to a non-null value for an existing installation will result in deregistering
```

Presumably the situation this comment is warning about doesn't happen during a first installation.

---

Yep, good point!